### PR TITLE
Making changes to the TS modules after an API update.

### PR DESCRIPTION
**Pull request by kiddinn**
This PR updates the Timesketch exporter and enhancer modules according to updates upstream in the API client.

It also ups the requirements of the aPI client that is needed.

Additionally a new recipe is added that uses the TS enhancer, as well as to enable the TS enhancer as a module.

### CODE REVIEW

File: data/recipes/upload_web_ts.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "upload_web_ts",
+    "description": "Uploads a parsed web history or EVTX records stored in a CSV or Plaso file to Timesketch.",
```
Comment (tomchop) [inline data/recipes/upload_web_ts.json:3]: Given that this is very similar to `upload_ts`, I'd try to stick to the other recipe's description and add "and uses web artifact analyzers" or something.

File: data/recipes/upload_web_ts.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "paths": "@files"
+        }
+    }, {
+        "wants": ["FilesystemCollector"],
+        "name": "TimesketchExporter",
+        "args": {
+            "incident_id": "@incident_id",
+            "token_password": "@token_password",
+            "sketch_id": "@sketch_id",
+	    "analyzers": "browser_search,browser_timeframe,account_finder,phishy_domains,evtx_gap,login,win_crash,safebrowsing,chain,"
```
Comment (tomchop) [inline data/recipes/upload_web_ts.json:18]: plz fix indentation

```suggestion
	      "analyzers": "browser_search,browser_timeframe,account_finder,phishy_domains,evtx_gap,login,win_crash,safebrowsing,chain,"
```

File: data/recipes/upload_web_ts.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        "name": "TimesketchEnhancer",
+        "args": {
+	    "wait_for_analyzers": "@wait_for_analyzers",
+            "searches_to_skip": "@searches_to_skip",
+            "aggregations_to_skip": "@aggregations_to_skip",
+            "include_stories": "@timesketch_include_stories",
+            "token_password": "@token_password",
+	    "max_checks": "@analyzer_max_checks",
+	    "formatter": "html"
+	}
```
Comment (tomchop) [inline data/recipes/upload_web_ts.json:31]:
```suggestion
            "token_password": "@token_password",
	      "max_checks": "@analyzer_max_checks",
	      "formatter": "html"
	}
```

File: data/recipes/upload_web_ts.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        ["files", "Comma-separated list of paths to CSV files or Plaso storage files", null],
+        ["--sketch_id", "Sketch to which the timeline should be added", null],
+        ["--token_password", "Optional custom password to decrypt Timesketch credential file with", ""],
+        ["--incident_id", "Incident ID (used for Timesketch description)", null],
+	["--wait_for_analyzers", "Wait for analyzers until they complete their run, if set to False the TS enhancer will be skipped", true],
+	["--timesketch_include_stories", "Include story dumps in reports", false],
+	["--searches_to_skip", "A comma separated list of saved searches that should not be uploaded.", null],
+	["--analyzer_max_checks", "Number of wait cycles (per cycle is 3 seconds) before terminating wait for analyzers to complete.", 0],
+	["--aggregations_to_skip", "A comma separated list of aggregation names that should not be uploaded.", null]
+    ]
```
Comment (tomchop) [inline data/recipes/upload_web_ts.json:43]: Were you mixing tabs and spaces? 🤔


```suggestion
    "args": [
        ["files", "Comma-separated list of paths to CSV files or Plaso storage files", null],
        ["--sketch_id", "Sketch to which the timeline should be added", null],
        ["--token_password", "Optional custom password to decrypt Timesketch credential file with", ""],
        ["--incident_id", "Incident ID (used for Timesketch description)", null],
        ["--wait_for_analyzers", "Wait for analyzers until they complete their run, if set to False the TS enhancer will be skipped", true],
        ["--timesketch_include_stories", "Include story dumps in reports", false],
        ["--searches_to_skip", "A comma separated list of saved searches that should not be uploaded.", null],
        ["--analyzer_max_checks", "Number of wait cycles (per cycle is 3 seconds) before terminating wait for analyzers to complete.", 0],
        ["--aggregations_to_skip", "A comma separated list of aggregation names that should not be uploaded.", null]
    ]
```

**Review by tomchop — CHANGES_REQUESTED** [review]

File: data/recipes/upload_web_ts.json (hunk 1 above)
Comment (kiddinn) [reply data/recipes/upload_web_ts.json]: reworded

**Review by kiddinn — COMMENTED** [review]
reworded

**Comment by kiddinn** [comment]
@tomchop PTAL

**Review by tomchop — APPROVED** [review]

**Pull request merged**